src/activities/bargame/Bargame.qml

```diff
@@ -0,0 +1,388 @@
+/* GCompris - bargame.qml
+ *
+ * Copyright (C) 2016 UTKARSH TIWARI <iamutkarshtiwari@kde.org>
+ *
+ * Authors:
+ *   Yves Combe (GTK+ version)
+ *   UTKARSH TIWARI <iamutkarshtiwari@kde.org > (Qt Quick port)
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.1
+import GCompris 1.0
+
+import "../../core"
+import "bargame.js" as Activity
+
+ActivityBase {
+    id: activity
+
+    onStart: focus = true
+    onStop: {}
+
+    pageComponent: Image {
+        id: rootWindow
+        source: "qrc:/gcompris/src/activities/bargame/resource/school_bg1.svg"
+        sourceSize.height: parent.height
+        sourceSize.width: parent.width
+        anchors.fill: parent
+        signal start
+        signal stop
+
+        Component.onCompleted: {
+            dialogActivityConfig.getInitialConfiguration()
+            activity.start.connect(start)
+            activity.stop.connect(stop)
+        }
+
+        // Add here the QML items you need to access in javascript
+        QtObject {
+            id: items
+            property Item main: activity.main
+            property Item boxModel: boxModel
+            property Item rootWindow: rootWindow
+            property alias tux: tux
+            property alias blueBalls: blueBalls
+            property alias greenBalls: greenBalls
+            property alias tuxArea: tuxArea
+            property alias boxes: boxes
+            property alias masks: masks
+            property alias numberLabel: numberLabel
+            property alias answerBallsPlacement: answerBallsPlacement
+            property alias answerBalls: answerBalls
+            property alias bar: bar
+            property alias bonus: bonus
+        }
```

Comment:
I would add mode here and use it in replacement of Activity.sublevel (the name corresponds more to what is the value and you'll get the advantages of qml bindings)